docs/xxcodecachetotalmaxrampercentage.md

```diff
@@ -0,0 +1,59 @@
+* which accompanies this distribution and is available at
+* https://www.eclipse.org/legal/epl-2.0/ or the Apache
+* License, Version 2.0 which accompanies this distribution and
+* is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* This Source Code may also be made available under the
+* following Secondary Licenses when the conditions for such
+* availability set forth in the Eclipse Public License, v. 2.0
+* are satisfied: GNU General Public License, version 2 with
+* the GNU Classpath Exception [1] and GNU General Public
+* License, version 2 with the OpenJDK Assembly Exception [2].
+*
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] http://openjdk.java.net/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+-->
+
+# -XX:codecachetotalMaxRAMPercentage
+
+This option sets the maximum limit for the total JIT code cache size.
+
+## Syntax
+
+        -XX:codecachetotalMaxRAMPercentage=<number>
+
+| Setting                 | Value      | Default                     |
+|-------------------------|------------|:---------------------------:|
+| `<number>`    | Between 1 and 100 (fractional numbers are allowed) |  25                       |
+
+Where `<number>` is the maximum limit that is expressed as a percentage of the available physical memory.
+
+For example, if you specify `-XX:codecachetotalMaxRAMPercentage=30`, the VM is not allowed to use more than 30% of the available physical memory for the internal JIT code caches.
+
+## Explanation
+
+The default total size of the JIT code cache is computed as the minimum of the following two limits:
+
+- A platform-based limit that is expressed as an absolute memory value. This limit is set at 256 MB for 64-bit systems and at 64 MB for 31/32-bit systems.
+- A percentage of the amount of available physical memory the VM process is allowed to use. By default, the VM is not allowed to use more than 25% of the available physical memory for its code caches.
+
+For example, in a 64-bit platform, the platform specific code cache limit is 256 MB. If the VM is constrained to less than 1024 MB (say 900 MB), then 25% of physical memory is less than 256 MB (the platform-based limit). Therefore, the default total size is calculated as a percentage of the amount of available physical memory (225 - minimum of the two limits).
+
+In environments with low physical memory availability, the VM might use too much of the available memory for code caches, leaving little memory for the VM or JIT data structures. With the `-XX:codecachetotalMaxRAMPercentage` option, you can reduce the default total size of the JIT code cache.
+
+If you want to increase the total code cache size as an absolute value (platform-based limit), you can use the [`-XX:codecachetotal`](xxcodecachetotal.md) or the [`-Xcodecachetotal`](xcodecachetotal.md) options.
```

Comment:
Another suggestion for rephrasing:

The default total size of the JIT code cache is computed as the minimum of the following two limits:

- A platform-based limit that is expressed as an absolute memory value. This limit is set at 256 MB for 64-bit systems and at 64 MB for 31/32-bit systems.
- A percentage of the amount of available physical memory the JVM process is allowed to use. By default, the JVM is not allowed to use more than 25% of the available physical memory for its code caches.

This second limit is relevant in memory constrained environments, as it prevents the JVM from using too much memory for its code caches, leaving too little memory for other VM/JIT data structures.

For example, on a 64-bit system, the platform specific code cache limit is 256 MB. If the JVM is constrained to less than 1024 MB, say 512 MB, then the code cache limit in effect becomes 128 MB (25% of 512 MB), because 128 MB is less than 256 MB (the platform-based limit).

To fine-tune the code cache size limit for your specific application use the `-XX:codecachetotalMaxRAMPercentage` option.
As an alternative, use the [`-XX:codecachetotal`](xxcodecachetotal.md) or the [`-Xcodecachetotal`](xcodecachetotal.md) options to set the code cache size limit as an absolute value (platform-based limit). Note that the absolute value specified by these options takes precedence over the percentage specified by the `-XX:codecachetotalMaxRAMPercentage` option.